src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -7,171 +7,193 @@
-
-        public override Symbol ContainingSymbol => ContainingType;
-
-        public override ImmutableArray<Location> Locations => ContainingType.Locations;
-
         public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray<SyntaxReference>.Empty;
 
-        public override Accessibility DeclaredAccessibility => Accessibility.Protected;
-
-        public override bool IsStatic => false;
-
-        public override bool IsVirtual => !IsOverride;
-
-        public override bool IsOverride { get; }
-
-        public override bool IsAbstract => false;
-
-        public override bool IsSealed => false;
+        IAttributeTargetSymbol IAttributeTargetSymbol.AttributesOwner => this;
 
-        public override bool IsExtern => false;
+        AttributeLocation IAttributeTargetSymbol.AllowedAttributeLocations => AttributeLocation.None;
 
-        internal override bool HasSpecialName => false;
+        AttributeLocation IAttributeTargetSymbol.DefaultAttributeLocation => AttributeLocation.None;
 
-        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+        protected override Location TypeLocation
+            => ContainingType.Locations[0];
 
-        internal override bool MustCallMethodsDirectly => false;
+        protected override SyntaxTokenList GetModifierTokens(SyntaxNode syntax)
+            => new SyntaxTokenList();
 
-        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+        public override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList
+            => new SyntaxList<AttributeListSyntax>();
 
-        public override string Name => PropertyName;
-
-        public override ImmutableArray<CSharpAttributeData> GetAttributes() => ImmutableArray<CSharpAttributeData>.Empty;
+        protected override void CheckForBlockAndExpressionBody(CSharpSyntaxNode syntax, DiagnosticBag diagnostics)
+        {
+            // Nothing to do here
+        }
 
-        private sealed class GetAccessorSymbol : SynthesizedInstanceMethodSymbol
+        protected override bool ReportProtectedMemberInSealedTypeError(Location location, DiagnosticBag diagnostics)
         {
-            private readonly SynthesizedRecordEqualityContractProperty _property;
+            return false;
+        }
 
-            public GetAccessorSymbol(SynthesizedRecordEqualityContractProperty property, string name)
+        protected override SourcePropertyAccessorSymbol? CreateAccessorSymbol(
+            bool isGet,
+            CSharpSyntaxNode? syntax,
+            PropertySymbol? explicitlyImplementedPropertyOpt,
+            string? aliasQualifierOpt,
+            bool isAutoPropertyAccessor,
+            bool isExplicitInterfaceImplementation,
+            DiagnosticBag diagnostics)
+        {
+            if (!isGet)
             {
-                _property = property;
-                Name = name;
+                return null;
```

Comment:
Can we reach this? If not, consider throwing as unreachable instead.

Reply:
> Can we reach this? If not, consider throwing as unreachable instead.

Yes, this method is called for both accessors, it is up to the method to decide if it should create one. 

---
In reply to: [452479312](https://github.com/dotnet/roslyn/pull/45831#discussion_r452479312) [](ancestors = 452479312)